File: schemas/team_case_history.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class TeamCaseHistoryCreate(BaseModel):
    case_semesters_id: str
    started_at: Optional[datetime] = None
    is_current: bool = True


class TeamCaseHistoryResponse(BaseModel):
    id: str
    team_id: str
    case_semesters_id: str
    case_id: Optional[str] = None
    case_title: Optional[str] = None
    semester_id: Optional[int] = None
    semester_season: Optional[str] = None
    semester_year: Optional[int] = None
    started_at: Optional[datetime] = None
    ended_at: Optional[datetime] = None
    is_current: bool
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

File: frontend/src/services/teamCaseHistory.ts
import api from './api';

export interface TeamCaseHistory {
  id: string;
  team_id: string;
  case_semesters_id: string;
  case_id: string;
  case_title: string;
  semester_id: number;
  semester_season: string;
  semester_year: number;
  started_at: string;
  ended_at: string | null;
  is_current: boolean;
  created_at: string;
  updated_at: string;
}

export interface AssignCaseData {
  case_semesters_id: string;
  started_at?: string;
  is_current?: boolean;
}

export const getTeamHistory = async (teamId: string): Promise<TeamCaseHistory[]> => {
  const response = await api.get(`/teams/${teamId}/history`);
  return response.data;
};

export const assignCaseToTeam = async (teamId: string, data: AssignCaseData): Promise<TeamCaseHistory> => {
  const response = await api.post(`/teams/${teamId}/history`, data);
  return response.data;
};

// Завершаем текущий активный кейс
export const endCurrentCase = async (teamId: string): Promise<TeamCaseHistory> => {
  const response = await api.post(`/teams/${teamId}/history/end-current`);
  return response.data;
};